[Bug]: get_hdf5io() does not work

### What happened?

A recent refactoring of `get_hdf5io` broke the code so now it raises an error on correct usage. There are no unit tests for this function.

### Steps to Reproduce

```shell
with get_hdf5io(self.path, "w") as io:
   pass
```


### Traceback

```shell
self = <tests.unit.common.test_common_io.TestGetHdf5IO testMethod=test_gethdf5io>

    def test_gethdf5io(self):
        """Test the get_hdf5io convenience method with manager=None."""
>       with get_hdf5io(self.path, "w") as io:

tests/unit/common/test_common_io.py:82: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/hdmf/utils.py:649: in func_call
    return func(**pargs)
src/hdmf/common/__init__.py:189: in get_hdf5io
    return HDF5IO.__init__(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = ()
kwargs = {'comm': None, 'driver': None, 'file': None, 'manager': <hdmf.build.manager.BuildManager object at 0x12f681d30>, ...}
pargs = {'comm': None, 'driver': None, 'file': None, 'manager': <hdmf.build.manager.BuildManager object at 0x12f681d30>, ...}

    def func_call(*args, **kwargs):
        pargs = _check_args(args, kwargs)
>       return func(args[0], **pargs)
E       IndexError: tuple index out of range

src/hdmf/utils.py:645: IndexError
```


### Operating System

macOS

### Python Executable

Python

### Python Version

3.9

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/hdmf-dev/hdmf/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 # HDMF Changelog
 
-## HDMF 3.5.0 (Upcoming)
+## HDMF 3.4.3 (September 14, 2022)
 
 ### Minor improvements
 - Began to deprecate the use of the testing script `test.py` in favor of `pytest` and `test_gallery.py`.
@@ -10,6 +10,7 @@
 ### Bug fixes
 - Fixed CI and flake8 issues. @rly ([#760](https://github.com/hdmf-dev/hdmf/pull/760))
 - Updated uses of pandas.DataFrame.set_index to avoid FutureWarnings for pandas >=1.5.x @oruebel ([#762](https://github.com/hdmf-dev/hdmf/pull/762))
+- Fixed broken `hdmf.common.get_hdf5io` function. @rly ([#765](https://github.com/hdmf-dev/hdmf/pull/765))
 
 ## HDMF 3.4.2 (August 26, 2022)
 

--- a/src/hdmf/common/__init__.py
+++ b/src/hdmf/common/__init__.py
@@ -181,12 +181,12 @@ def validate(**kwargs):
 @docval(*get_docval(HDF5IO.__init__), is_method=False)
 def get_hdf5io(**kwargs):
     """
-    A convenience method for getting an HDF5IO object
+    A convenience method for getting an HDF5IO object using an HDMF-common build manager if none is provided.
     """
     manager = getargs('manager', kwargs)
     if manager is None:
         kwargs['manager'] = get_manager()
-    return HDF5IO.__init__(**kwargs)
+    return HDF5IO(**kwargs)
 
 
 # load the hdmf-common namespace

--- a/tests/unit/common/test_common_io.py
+++ b/tests/unit/common/test_common_io.py
@@ -87,4 +87,3 @@ class TestGetHdf5IO(TestCase):
         manager = get_manager()
         with get_hdf5io(self.path, "w", manager=manager) as io:
             self.assertIs(io.manager, manager)
-

--- a/tests/unit/common/test_common_io.py
+++ b/tests/unit/common/test_common_io.py
@@ -1,7 +1,7 @@
 from h5py import File
 
 from hdmf.backends.hdf5 import HDF5IO
-from hdmf.common import Container, get_manager
+from hdmf.common import Container, get_manager, get_hdf5io
 from hdmf.spec import NamespaceCatalog
 from hdmf.testing import TestCase, remove_test_file
 
@@ -67,3 +67,24 @@ class TestCacheSpec(TestCase):
                             self.assertIsNotNone(cached_spec)
                         with self.subTest('Cached spec matches original spec'):
                             self.assertDictEqual(original_spec, cached_spec)
+
+
+class TestGetHdf5IO(TestCase):
+
+    def setUp(self):
+        self.path = get_temp_filepath()
+
+    def tearDown(self):
+        remove_test_file(self.path)
+
+    def test_gethdf5io(self):
+        """Test the get_hdf5io convenience method with manager=None."""
+        with get_hdf5io(self.path, "w") as io:
+            self.assertIsNotNone(io.manager)
+
+    def test_gethdf5io_manager(self):
+        """Test the get_hdf5io convenience method with manager set."""
+        manager = get_manager()
+        with get_hdf5io(self.path, "w", manager=manager) as io:
+            self.assertIs(io.manager, manager)
+

